sdk/src/main/scala/com/horizen/account/forger/AccountForgeMessageBuilder.scala

```diff
@@ -77,6 +78,41 @@ class AccountForgeMessageBuilder(mainchainSynchronizer: MainchainSynchronizer,
     }
   }
 
+  def calculateBaseFee(nodeView: View, parentId: BlockId): BigInteger = {
+    // If the current block is the first block, return the InitialBaseFee.
+    if (nodeView.history.getCurrentHeight < 2) {
+      return Account.INITIAL_BASE_FEE
+    }
+
+    nodeView.history.getBlockById(parentId).asScala match {
+    case None => Account.INITIAL_BASE_FEE
+    case Some(block) =>
+      val blockHeader = block.header
+      val gasTarget = blockHeader.gasLimit / 2
+
+        // If the parent gasUsed is the same as the target, the baseFee remains unchanged
+      if (blockHeader.gasUsed == gasTarget) {
+        return blockHeader.baseFee
+      }
+
+      if (blockHeader.gasUsed > gasTarget) {
+        // If the parent block used more gas than its target, the baseFee should increase by 12.5%
+        var baseFeeInc: BigInteger = BigInteger.valueOf(blockHeader.gasUsed - gasTarget)
+        baseFeeInc = baseFeeInc.multiply(blockHeader.baseFee)
+        baseFeeInc = baseFeeInc.divide(BigInteger.valueOf(gasTarget))
+        baseFeeInc = baseFeeInc.divide(BigInteger.valueOf(8))
+        blockHeader.baseFee.add(baseFeeInc.max(BigInteger.ONE))
+      } else {
+        // Otherwise if the parent block used less gas than its target, the baseFee should decrease by 12.5%
+        var baseFeeDec: BigInteger = BigInteger.valueOf(gasTarget - blockHeader.gasUsed)
+        baseFeeDec = baseFeeDec.multiply(blockHeader.baseFee)
+        baseFeeDec = baseFeeDec.divide(BigInteger.valueOf(gasTarget))
+        baseFeeDec = baseFeeDec.divide(BigInteger.valueOf(8))
+        blockHeader.baseFee.subtract(baseFeeDec).max(BigInteger.ONE)
```

Comment:
Should be `BigInteger.ZERO` here, according to GETH implementation.

Reply:
I know, we discussed to change this later to another value, but I just wanted to prevent it to become 0 for now.